java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -0,0 +1,420 @@
+     * @param setCondition Set the value only if the given condition is met (within the key or path).
+     * @return A simple <code>"OK"</code> response if the value is successfully set. If value isn't
+     *     set because of <code>setCondition</code>, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * String value = client.Json.set(client, "doc", , ".", "{'a': 1.0, 'b': 2}", ConditionalChange.ONLY_IF_DOES_NOT_EXIST).get();
+     * assert value.equals("OK");
+     * }</pre>
+     */
+    public static CompletableFuture<String> set(
+            @NonNull BaseClient client,
+            @NonNull String key,
+            @NonNull String path,
+            @NonNull String value,
+            @NonNull ConditionalChange setCondition) {
+        return executeCommand(
+                client, new String[] {JSON_SET, key, path, value, setCondition.getValkeyApi()});
+    }
+
+    /**
+     * Sets the JSON value at the specified <code>path</code> stored at <code>key</code>.
+     *
+     * @param client The Valkey GLIDE client to execute the command.
+     * @param key The <code>key</code> of the JSON document.
+     * @param path Represents the path within the JSON document where the value will be set. The key
+     *     will be modified only if <code>value</code> is added as the last child in the specified
+     *     <code>path</code>, or if the specified <code>path</code> acts as the parent of a new child
+     *     being added.
+     * @param value The value to set at the specific path, in JSON formatted GlideString.
+     * @param setCondition Set the value only if the given condition is met (within the key or path).
+     * @return A simple <code>"OK"</code> response if the value is successfully set. If value isn't
+     *     set because of <code>setCondition</code>, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * String value = client.Json.set(client, gs("doc"), , gs("."), gs("{'a': 1.0, 'b': 2}"), ConditionalChange.ONLY_IF_DOES_NOT_EXIST).get();
+     * assert value.equals("OK");
+     * }</pre>
+     */
+    public static CompletableFuture<String> set(
+            @NonNull BaseClient client,
+            @NonNull GlideString key,
+            @NonNull GlideString path,
+            @NonNull GlideString value,
+            @NonNull ConditionalChange setCondition) {
+        return executeCommand(
+                client,
+                new GlideString[] {gs(JSON_SET), key, path, value, gs(setCondition.getValkeyApi())});
+    }
+
+    /**
+     * Retrieves the JSON value at the specified <code>path</code> stored at <code>key</code>.
+     *
+     * @param client The Valkey GLIDE client to execute the command.
+     * @param key The <code>key</code> of the JSON document.
+     * @param path The path within the JSON document.
+     * @return For JSONPath (path starts with <code>$</code>): Returns a stringifies JSON list of
```

Comment:
We should format this like we do for python.  
```
<ul>
<li>For JSONPath (path starts with <code>$</code>): Returns a stringified JSON list of bytes replies for every possible path, or a byte string representation of an empty array, if path doesn't exist. 
<li>For legacy path (path doesn't start with <code>$</code>): Returns a byte string representation of the value in <code>path</code>. If <code>path</code> doesn't exist, an error is raised. If <code>key</code> doesn't exist, returns <code>null</code>.
</ul>
```
or just copy-paste the python equivalent and change None to null.